### PR TITLE
Json poc

### PR DESCRIPTION
**Pull request by albertmink**
#4861

Alignement to abapGit repo
- [x] [zcl_abapgit_ajson* is used by mirrored objects. To avoid confusion, please use a different prefix. thanks
](https://github.com/abapGit/abapGit/pull/5126#issuecomment-966412589) see 539964ddd6222c2e

### CODE REVIEW

**Comment by mbtools** [comment]
zcl_abapgit_ajson* is used by mirrored objects. To avoid confusion, please use a different prefix. thanks

**Comment by larshp** [comment]
`zif_abapgit_aff_chko_v1.intf.abap` copy the file from https://github.com/abapGit/aff_mirror/tree/main/src, it should be okay there. Later we will setup mirroring from that repo into abapGit

**Comment by albertmink** [comment]
we need more INTF from the AFF, no automation yet

**Comment by albertmink** [comment]
Can't wait for Quick-Fixing this kind of findings 🙃 
image

File: src/objects/zcl_abapgit_object_chko.clas.abap [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    CREATE DATA lr_chko TYPE ('ZIF_ABAPGIT_AFF_CHKO_V1=>TY_MAIN').
+    ASSIGN lr_chko->* TO <chko_agit>.
+
+    IF <chko_agit> IS NOT ASSIGNED.
+      RETURN.
+    ENDIF.
+
+    CREATE OBJECT object_ajson TYPE ('ZCL_ABAPGIT_AJSON_CNT_HANDLER').
+
+    json_as_xstring = mo_files->read_raw( iv_ext = 'json' ) ##NO_TEXT.
```
[the remark this answers is not in the document]
Comment (larshp) [reply src/objects/zcl_abapgit_object_chko.clas.abap:108]: hmm, bad bot

**Comment by larshp** [comment]
https://github.com/abaplint/abaplint/issues/2239 opened for the quickfix

File: src/objects/zcl_abapgit_object_chko.clas.abap (hunk 1 above)
[the remark this answers is not in the document]
Comment (larshp) [reply src/objects/zcl_abapgit_object_chko.clas.abap:108]: note, the rule is a thing in abapGit, we dont like the NO_TEXT pragma, as we cannot move texts to the correct places anyhow

File: src/objects/zcl_abapgit_object_chko.clas.abap [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~exists.
+
+    DATA chko_db_api TYPE REF TO object.
+    DATA name TYPE c LENGTH 30.
+
+    name = ms_item-obj_name .
```
[the remark this answers is not in the document]
Comment (larshp) [reply src/objects/zcl_abapgit_object_chko.clas.abap:143]: hmm, it seems broken as such, will take a look later today

**Comment by larshp** [comment]
unused types: update the config

**Review by schneidermic0 — COMMENTED** [review]
I have just added some small comments.

**Comment by mbtools** [comment]
looks like it's coming along nicely 😃. 

let's keep `/src/objects` for the object types only. suggestion:
- move `zcl_abapgit_json_cnt_handler` to `/src/objects/json`
- move `/src/aff` to `/src/objects/aff`

**Comment by albertmink** [comment]
I agree on the folder topic. But keeping in mind that it is ABAP, than from the available 30 chars there would be 23 reserved due to folder convention?
`zif_abapgit_objects_aff` respectively `zif_abapgit_objects_json` contains already 23 character.

Any suggestion on that @mbtools ? Maybe drop the __objects__ from the object name

**Comment by albertmink** [comment]
navigation to CHKO object seems broken?
image

**Comment by albertmink** [comment]
serialization of CHKO here [#32dddca61057b95a852dcd1e362e2896740924cb](https://github.com/albertmink/testing_chko/commit/32dddca61057b95a852dcd1e362e2896740924cb)
It kind of works, but still need the empty line as end-of-file marker.

**Comment by schneidermic0** [comment]
We would like to check whether we can remove some domain specific code for CHKO and replace it with generic code using ABAP file formats (AFF). I see the option to get rid of some dynamic code for domain owners (e.g, code in https://github.com/abapGit/abapGit/blob/0962c9aa7caed17e6f3d8d34f6df5c5a20fa69cf/src/objects/zcl_abapgit_object_chko.clas.locals_imp.abap). This domain specific code is basically available in the domain-specific implementation and is just called by the generic AFF implementation. The generic AFF implementation must be dynamic to be syntactically correct also in releases where AFF is not available. 

For new object types, like CHKO, this should be fine.

**Comment by schneidermic0** [comment]
> serialization of CHKO here #32dddca61057b95a852dcd1e362e2896740924cb
It kind of works, but still need

Is the file `zma_chko_aff.chko.xml` expected?

**Comment by mbtools** [comment]
Some housekeeping items:

- [ ] Descriptions of new packages/classes/interfacees should begin with `abapGit - ...`, except for `zcl_abapgit_object_chko` where we it should be `CHKO - ATC Check Object`
- [ ] Add minimal test case to repo https://github.com/abapGit-tests/CHKO

Sidenote: As you know CHKO isn't available in 754 (ABAP Trial 1909) which limits who will be able to test any of this. I suggest adding another object type that's available in 754 as the next step.

**Comment by mbtools** [comment]
please remove the changes to `zcl_abapgit_objects` from this PR. we already merged a solution for this part with #5254

**Comment by larshp** [comment]
close this PR? perhaps take a local backup with some of the code

**Pull request closed without merging**